fintech-examples/fintech-ui/src/app/interceptors/auth.interceptor.ts

```diff
@@ -1,46 +1,54 @@
-import {HttpEvent, HttpHandler, HttpInterceptor, HttpRequest} from '@angular/common/http';
-import {Injectable} from '@angular/core';
-import {Observable, throwError} from 'rxjs';
-import {catchError} from 'rxjs/operators';
+import { HttpEvent, HttpHandler, HttpInterceptor, HttpRequest, HttpResponse } from '@angular/common/http';
+import { Injectable } from '@angular/core';
+import { Observable, throwError } from 'rxjs';
+import { tap } from 'rxjs/operators';
 import * as uuid from 'uuid';
 import { HeaderConfig } from '../models/consts';
-import { StorageService } from '../services/storage.service';
 import { AuthService } from '../services/auth.service';
-import {Router} from '@angular/router';
+import { Router } from '@angular/router';
 
 @Injectable()
 export class AuthInterceptor implements HttpInterceptor {
-  constructor(private router: Router, private storageService: StorageService, private authService: AuthService) {}
+  constructor(private router: Router, private authService: AuthService) {
+  }
 
   intercept(request: HttpRequest<any>, next: HttpHandler): Observable<HttpEvent<any>> {
     return this.handleRequest(request, next).pipe(
-      catchError(errors => {
-        return throwError(errors);
-      })
+      tap(response => {
+          if (response instanceof HttpResponse) {
+            const maxAge = response.headers.get(HeaderConfig.HEADER_FIELD_X_MAX_AGE);
+            if (maxAge !== null) {
+              this.authService.extendSessionAge(parseInt(maxAge, 0));
+            }
+          }
+        },
+        error => {
+          if (error.status === 401 /*HTTP_STATUS_UNAUTHORIZED*/) {
+            console.log('status was 401');
+            this.router.navigate(['/session-expired']);
```

Comment:
This navigation has to be handle in service because it's used in multiple places. We should have something like
`this.xxxxService.openSessionExpired`

Reply:
Currently its in two places. I think this is ok. In error handler and navbar component.